zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/validator/DocumentValidator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.validator;
+
+import static io.camunda.zeebe.gateway.rest.validator.RequestValidator.createProblemDetail;
+
+import io.camunda.zeebe.gateway.protocol.rest.DocumentMetadata;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Optional;
+import org.springframework.http.ProblemDetail;
+
+public class DocumentValidator {
+
+  public static Optional<ProblemDetail> validateDocumentMetadata(
+      DocumentMetadata metadata) {
+
+    final var violations = new ArrayList<String>();
+
+    if (metadata == null) {
+      return Optional.empty();
+    }
+
+    if (metadata.getFileName() != null && metadata.getFileName().isBlank()) {
+      violations.add("The file name must not be empty, if present");
+    }
+
+    if (metadata.getExpiresAt() != null) {
+      String expiresAt = metadata.getExpiresAt();
+      var dateTime = DateTimeFormatter.ISO_DATE_TIME;
+      try {
+        dateTime.parse(expiresAt);
+      } catch (Exception e) {
+        violations.add("The expiration date must be in ISO-8601 format");
+      }
```

Comment:
🔧 There already is a `RequestValidator::validateDate` method you can reuse here.